### PR TITLE
WIP feat: support MongoDB for claims storage

### PR DESCRIPTION
**Pull request by technosophos**
This is a trivial implementation of a MongoDB claim storage.

- [x] Main interface
- [x] Tests
- [x] Exposed to user via env var

### CODE REVIEW

**Comment by technosophos** [comment]
rebased

**Comment by technosophos** [comment]
I currently only have Fridays to work on this, so it's taking me a long time. I wanted to test it on both MongoDB and CosmosDB before taking off the WIP flag.

**Comment by technosophos** [comment]
This is now blocked until https://github.com/deislabs/cnab-go/pull/15 is merged.

File: pkg/utils/crud/mongodb.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		session:    session,
+		collection: session.DB(dbn).C(MongoClaimsCollection),
+		dbName:     dbn,
+	}, nil
+}
+
+func (s *mongoDBStore) List() ([]string, error) {
+	var res []doc
+	if err := s.collection.Find(nil).All(&res); err != nil {
+		return []string{}, wrapErr(err)
```
Comment (silvin-lubecki) [inline pkg/utils/crud/mongodb.go]: nit: I think it is safe here to return a nil value
```go
return nil, wrapErr(err)
```

File: pkg/utils/crud/mongodb.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		}
+		return []byte{}, wrapErr(err)
+	}
+	return res.Data, nil
+}
+func (s *mongoDBStore) Delete(name string) error {
+	return wrapErr(s.collection.Remove(map[string]string{"name": name}))
+}
+
+func wrapErr(err error) error {
```
Comment (silvin-lubecki) [inline pkg/utils/crud/mongodb.go]: What about using the [github.com/pkg/errors.Wrap](https://godoc.org/github.com/pkg/errors#Wrap) package instead ?

File: pkg/utils/crud/mongodb.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		dbName:     dbn,
+	}, nil
+}
+
+func (s *mongoDBStore) List() ([]string, error) {
+	var res []doc
+	if err := s.collection.Find(nil).All(&res); err != nil {
+		return []string{}, wrapErr(err)
+	}
+	buf := []string{}
```
Comment (silvin-lubecki) [inline pkg/utils/crud/mongodb.go]: nit: As we already know the buf size, we can reserve the capacity:
```go
buf := make([]string, 0, len(res))
```

File: pkg/utils/crud/mongodb.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	for _, v := range res {
+		buf = append(buf, v.Name)
+	}
+	return buf, nil
+}
+
+func (s *mongoDBStore) Store(name string, data []byte) error {
+	return wrapErr(s.collection.Insert(doc{name, data}))
+}
+func (s *mongoDBStore) Read(name string) ([]byte, error) {
```
Comment (silvin-lubecki) [inline pkg/utils/crud/mongodb.go]: nit: empyt line

**Comment by technosophos** [comment]
I'll update this PR as soon as I get the CRUD storage fixed on the cnab-go library.

**Comment by technosophos** [comment]
Closing this. I'll have to write a new version based on the changes I made on cnab-go.

**Pull request closed without merging**